src/common/snippets/src/lowered/port_descriptor.cpp

```diff
@@ -48,9 +50,22 @@ const VectorDims& PortDescriptor::get_shape() const {
 
 void PortDescriptor::set_shape(const VectorDims& tensor) {
     OPENVINO_ASSERT(m_tensor_shape, "Failed to set_shape: Tensor Shape is nullptr");
+    OPENVINO_ASSERT(m_subtensor_shape.size() <= m_tensor_shape->size(),
+                    "Snippets tensor descriptor: Subtensor shape must be less than or equal to tensor shape");
```

Comment:
This check doesn't verify passed `tensor`. I mean if I pass `[1,2,3]` to this method while the original `m_tensor_shape` is `[1,2,3,4,5]`, it should throw exception. But the current implementation will do nothing 🤔 
Should we check that `tensor.size() == m_tensor_shape->size()` here?

Reply:
I have changed it to compare with tensor size instead: `m_tensor_shape->size()` -> `tensor.size()`

> Should we check that `tensor.size() == m_tensor_shape->size()` here?

As for this, I'm not sure. It seems that the fact that they are not equal is being used in the code base.